File: Library/Models/Artist.cs
using Library.Providers;

namespace Library.Models;

public class Artist : Music
{
    private List<Song>? songs;

    public Artist(string id)
    {
        this.id = id;
    }

    public List<Song> Songs
    {
        get
        {
            if (songs == null)
            {
                songs = new();
                foreach (var song in Provider.Songs)
                {
                    foreach (var artist in song.Artists)
                    {
                        if (artist.Id == null) continue;
                        if (artist.Id.Equals(id)) songs.Add(song); continue;
                    }
                }
            }
            return songs;
        }
    }
}

File: Music.Client.Android/Java/short/app/src/main/java/com/zitherharpmusic/zhmshort/model/Artist.java
package com.zitherharpmusic.zhmshort.model;

import java.util.ArrayList;
import java.util.List;

public class Artist extends Music {
    private final String playlistId = "";
    private List<Song> songs;
    private List<Video> videos;

    public Artist(String id) {
        this.id = id;
    }

    public String getPlaylistUrl() {
        return "https://www.youtube.com/playlist?list=" + playlistId;
    }

    public List<Song> getSongs() {
        if (songs == null) {
            songs = new ArrayList<>();
            for (Song song : MusicProvider.getSongs()) {
                for (Artist artist : song.getArtists()) {
                    if (artist.getId().equals(id)) {
                        songs.add(song);
                    }
                }
            }
        }
        return songs;
    }

    public List<Video> getVideos() {
        if (videos == null) {
            videos = new ArrayList<>();
            for (Video video : MusicProvider.getVideos()) {
                for (Artist artist : video.getArtists()) {
                    if (artist.getId().equals(id)) {
                        videos.add(video);
                    }
                }
            }
        }
        return videos;
    }
}
